### ISSUE TITLE
crashes in Electron 14+ renderer when reloading

### ISSUE THREAD

**Issue by marcelblum** [almoghamdani/audify#33]
I think this must be related to Electron's deprecation of `app.allowRendererProcessReuse` and [related changes](https://github.com/electron/electron/issues/18397) introduced with Electron 14. I'm using Audify in Electron in the renderer process. But in Electron versions later than 13, if I reload the renderer or navigate away to a different page after having already created an instance of Audify, the renderer crashes with error -36861 if I then try to instantiate Audify again after navigating back to the original page or reloading it. The following works in Electron <=13 with `app.allowRendererProcessReuse=false` but crashes in 14+:
```javascript
const audify = require("audify");
location.reload();
//...
const audify = require("audify"); //crash!
```

My guess is that some kind of proper cleanup/unload function in Audify could resolve this. Any ideas as to how to approach this or what specifically could lead to this crash that would need to be cleaned up? In the renderer I've tried trickery like this but it doesn't prevent the crash:
```javascript
const audify = require("audify"); 
audify = null;
gc();
webContents.reloadIgnoringCache();
//...
const audify = require("audify"); //still crashes here
```

~~Also, interestingly, this happens:~~
<strike>
```javascript
BrowserWindow.loadFile("somepage.html");
//...
const audify = require("audify");
BrowserWindow.loadFile("differentpage.html");
//...
const audify = require("audify"); //no crash!
BrowserWindow.loadFile("somepage.html");
//...
const audify = require("audify"); //crash when tried again back at the first page
```
Audify can be instantiated multiple times over different renderer page loads without crashing, but if I try to instantiate it more than once during different "visits" to the same page in the same renderer session, it will crash, suggesting Electron is keeping a different v8 context for each page, and caching or faulty garbage cleanup is to blame for the crash.
</strike>
**Update**: Strike the above, I'm also sometimes getting crashes when doing the above on different pages, albeit less consistently than after doing a simple `location.reload()`.

**Comment by marcelblum** [comment]
Well this works around the issue but I don't recommend it 🤷‍♂️ 😈 
```js
const audify = require("audify");
webContents.forcefullyCrashRenderer();
webContents.reload();
//...
const audify = require("audify"); //no crash, because a new renderer process has been forced
```

**Comment by marcelblum** [comment]
I've found that the crashes on reload are eliminated if I comment out the following lines:
https://github.com/almoghamdani/audify/blob/6761e1259212948d4b6333106da654f366175e41/src/rt_audio.cpp#L109-L110
https://github.com/almoghamdani/audify/blob/6761e1259212948d4b6333106da654f366175e41/src/opus_decoder.cpp#L11-L12
https://github.com/almoghamdani/audify/blob/6761e1259212948d4b6333106da654f366175e41/src/opus_encoder.cpp#L13-L14
But otherwise the lib continues to work fine AFAICT (have not tested extremely thoroughly). I wonder why it is necessary to prevent destruction of the constructors like this? Maybe there is some use case scenario where this is a requirement?

**Comment by marcelblum** [comment]
Looks like adding `constructor.Reset()` to the destructor functions prevents the crash.

**Comment by almoghamdani** [comment]
Fixed in #34. Thanks for the PR 😄
